lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -246,31 +247,41 @@ sub _get_dirty_status {
 }
 
 # Obs version is parsed from files_iso.lst, which is updated from ObsRsync Gru tasks
-sub _get_version_in_folder {
+sub _get_builds_in_folder {
     my ($folder) = @_;
     open(my $fh, '<', Mojo::File->new($folder, $files_iso_filename)) or return "";
-    my $version;
+    my %seen;
     while (my $row = <$fh>) {
         chomp $row;
         next unless $row;
-        next if substr($row, 0, 1) eq "#";
-        if ($row =~ m/Build((\d)+\.(\d)+(\.(\d)+)?)/ or $row =~ m/Snapshot((\d)+(\.(\d)+)*)/) {
-            $version = $1;
-            last;
-        }
+        next unless ($row =~ m/Build((\d)+\.(\d)+(\.(\d)+)?)/ or $row =~ m/Snapshot((\d)+(\.(\d)+)*)/);
+        my $build = $1;
+        $seen{$build} = 1;
     }
     close $fh;
-    return $version;
+    return sort { $b cmp $a } keys %seen;
 }
 
-# Obs version is parsed from files_iso.lst, which is updated from ObsRsync Gru tasks
-sub _get_obs_version {
-    my ($c, $project) = @_;
-    my $helper = $c->obs_rsync;
-    my $home   = $helper->home;
-    (my $batch, $project) = $helper->get_first_batch($project);
+# Obs builds are parsed from files_iso.lst, which is updated from ObsRsync Gru tasks
+sub _get_obs_builds_text {
+    my ($c, $project, $last) = @_;
+    my $helper    = $c->obs_rsync;
+    my $home      = $helper->home;
+    my $subfolder = $last ? '.run_last' : '';
+    my %seen;
+    my $sub = sub {
+        my ($_project, $_batch) = @_;
+        my @_builds = _get_builds_in_folder(Mojo::File->new($home, $_project, $_batch, $subfolder));
```

Comment:
I wouldn't prefix a local variable with a `_`.

Reply:
Agreed.

Reply:
There are some good rules to follow for variable naming, but `$_name` is mostly unused so far. Personally i only use a) `$name` for normal sub scoped variables, b) `$NAME` for package scope variables, and c) `$Name` for package scope variables that are accessed from the outside (`our $Name = ...;`).

Reply:
For completeness, some people in the Perl community have started using `$_name` for exportable method modules. But that's a very fringe use case.
```
use Test::Mojo;
use Test::SomeModule '$_some_test_method';

my $t = Test::Mojo->new('MyApp');
$t->get_ok('/foo')->$_some_test_method(...)->status_is(200);
```

Reply:
So i'd suggest simply not using that naming scheme. 😉

Reply:
Your rules don't cover case when I want to distinguish variables defined inside sub from those which are defined outside of sub, but referenced from inside.

Reply:
Those aren't my rules, those are Perl community rules. But you're right, there are no special rules for your case. And i can't really think of a good reason for why anyone would want to keep track of that.

The only thing to watch out for with closures is circular references, but special naming conventions don't really help with that.